src/openms/source/FORMAT/MSPFile.cpp

```diff
@@ -148,9 +128,13 @@ namespace OpenMS
         line.split(' ', split);
         split[1].split('/', split2);
         String peptide = split2[0];
-        Int charge = split2[1].toInt();
-        // remove damn (O), also defined in 'Mods=' comment
-        peptide.substitute("(O)", "");
+        // in newer NIST versions, the charge is followed by the modification(s) e.g. "_1(0,A,Acetyl)"
+        vector<String> split3;
+        split2[1].split('_', split3);
+        Int charge = split3[0].toInt();
+
+        // remove modifications inside the peptide string, since it is also defined in 'Mods=' comment
```

Comment:
do both carry the same information?

Reply:
yes

Reply:
I don't know where this other format comes from